DuckDuckGo/BookmarksViewController.swift

```diff
@@ -357,6 +357,10 @@ class BookmarksViewController: UITableViewController {
     }
 
     func importBookmarks(fromHtml html: String) {
+        // pre-emptively deregisterForNotifications so that bookmarksCachingSearch is not saturated with notification events
+        // and constantly rebuilding while bookmarks are being imported (bookmark files could be very large)
+        bookmarksCachingSearch.deregisterForNotifications()
```

Comment:
Very very minor, but I'd prefer if this was moved from the view controller since it feels like something the view controller shouldn't have to know about. E.g. managed by BookmarksImporter or even just behind a method on bookmarksCachingSearch (like `prepareForImport()` or something).

Reply:
Also wondering does the import trigger loads of notifications in any other bookmarksCachingSearch instances that are around? I mean it won't right now since I think the same shared object is used everywhere? But in future it would be easy for someone to not release this and trip up 

How about importing firing notifications before starting and on completion that bookmarksCachingSearch can listen to and then manage what it does with that information itself?

Reply:
> Also wondering does the import trigger loads of notifications in any other bookmarksCachingSearch instances that are around? I mean it won't right now since I think the same shared object is used everywhere? But in future it would be easy for someone to not release this and trip up
> 
> How about importing firing notifications before starting and on completion that bookmarksCachingSearch can listen to and then manage what it does with that information itself?

Sounds great + I've done it this way :slightly_smiling_face: